src/OfflineWorkflowsSample/OfflineWorkflowsSample/ViewModels/ItemPages/Offline/DownloadMapAreaViewModel.cs

```diff
@@ -0,0 +1,422 @@
+﻿using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Portal;
+using Esri.ArcGISRuntime.Symbology;
+using Esri.ArcGISRuntime.Tasks.Offline;
+using Esri.ArcGISRuntime.UI;
+using OfflineWorkflowsSample.Infrastructure;
+using OfflineWorkflowsSample.Models;
+using Prism.Commands;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Windows.Storage;
+using Windows.System;
+using Esri.ArcGISRuntime.Geometry;
+using OfflineWorkflowsSample.Infrastructure.ViewServices;
+using Prism.Windows.Mvvm;
+using Prism.Windows.Navigation;
+
+namespace OfflineWorkflowsSample.DownloadMapArea
+{
+    public class DownloadMapAreaViewModel : ViewModelBase
+    {
+        #region Sample code
+
+        private async Task ConfigureMap()
+        {
+            _areasOverlay = new GraphicsOverlay();
+
+            // Load map from portal
+            await Map.LoadAsync();
+        }
+
+        private async void QueryMapAreas()
+        {
+            try
+            {
+                // Clear existing overlays
+                _mapViewService.ClearOverlays();
+
+                // Clear existing areas (in case user wants to refresh results)
+                MapAreas.Clear();
+
+                // Create new task to 
+                var offlineMapTask = await OfflineMapTask.CreateAsync(Map);
+
+                // Get list of areas
+                IReadOnlyList<PreplannedMapArea> preplannedMapAreas = await offlineMapTask.GetPreplannedMapAreasAsync();
+
+                // Create UI from the areas
+                foreach (var preplannedMapArea in preplannedMapAreas.OrderBy(x => x.PortalItem.Title))
+                {
+                    // Load area to get the metadata 
+                    await preplannedMapArea.LoadAsync();
+                    // Using a custom model for easier visualization
+                    var model = new MapAreaModel(preplannedMapArea);
+                    MapAreas.Add(model);
+                    // Graphic that shows the area in the map
+                    var graphic = new Graphic(preplannedMapArea.AreaOfInterest, GetSymbolForColor(model.DisplayColor));
+                    graphic.Attributes.Add("Name", preplannedMapArea.PortalItem.Title);
+                    _areasOverlay.Graphics.Add(graphic);
+                }
+
+                if (!preplannedMapAreas.Any())
+                {
+                    await _windowService.ShowAlertAsync("No preplanned map areas available.");
+                }
+                else
+                {
+                    // Show the overlays on the map
+                    _mapViewService.AddGraphicsOverlay(_areasOverlay);
```

Comment:
Instead of doing it this way, why not just bind a GraphicsOverlayCollection on to the MapView?